components/esp_modem/test/target_ota/components/manual_ota/manual_ota.hpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Unlicense OR CC0-1.0
+ */
+#pragma once
+#include <vector>
+#include "esp_http_client.h"
+#include "esp_partition.h"
+#include "esp_transport_tcp.h"
+#include "esp_ota_ops.h"
+
+class manual_ota {
+public:
+    /**
+     * @brief Set the preferred mode
+     */
+    enum class mode {
+        BATCH,      /**< Read data chunk from TCP and pass it to SSL, restore session on reconnection */
+        NORMAL      /**< Use standard partial download, continuously passing data from TCP to mbedTLS */
+    } mode_ {mode::BATCH};
+
+    /**
+     * @brief Set the OTA batch size in kB
+     *
+     * This would allocate two big buffers:
+     * - one for reading from TCP socket and
+     * - one for passing to mbedTLS for description
+     */
+    size_t size_{32};
+
+    /**
+     * @brief Set timeout in seconds
+     *
+     * This is the network timeout, so if less data than the batch size received
+     * the timeout (and no EOF) we should proceed with passing the data to mbedtls
+     */
+    int timeout_{2};
+
+    /**
+     * @brief Set common name of the server to verify
+     */
+    const char *common_name_;
+    /**
+     * @brief Wrapper around the http client -- Please set the http config
+     */
+    class http_client {
+        friend class manual_ota;
+        http_client() {}
```

Comment:
Nitpick: This constructor could be defaulted or completely removed (compiler will default it).